packages/jsii/bin/jsii.ts

```diff
@@ -57,12 +57,17 @@ const warningTypes = Object.keys(enabledWarnings);
 
   const compiler = new Compiler({
     projectInfo,
-    watch: argv.watch,
     projectReferences: argv['project-references'],
     failOnWarnings: argv['fail-on-warnings'],
   });
 
-  return { projectRoot, emitResult: await compiler.emit() };
+  compiler.on('statusChanged', diagnostic => utils.logDiagnostic(diagnostic, projectRoot));
+
+  const result = argv.watch
+    // When watching, return a promise that never resolves... #magic
```

Comment:
why isn't this hidden inside `compiler.watch()`. That's how I'd expect it to behave without wrapping it with #magic

Reply:
That makes `watch`... impossible to test...